header-rewrite-filter/header_processor.cc

```diff
@@ -212,11 +212,13 @@ namespace HeaderRewriteFilter {
     }
 
     absl::Status SetBoolProcessor::parseOperation(std::vector<absl::string_view>& operation_expression, std::vector<absl::string_view>::iterator start) {
-        if (operation_expression.size() < Utility::SET_BOOL_MIN_NUM_ARGUMENTS) {
-            return absl::InvalidArgumentError("not enough arguments for set-bool");
-        }
-
         try {
+            if (operation_expression.size() < Utility::SET_BOOL_MIN_NUM_ARGUMENTS) {
+                return absl::InvalidArgumentError("not enough arguments for set-bool");
+            }
+
+            const bool isRequest = (operation_expression.at(0) == "http-request");
```

Comment:
Nit: should use `Utility::HTTP_REQUEST`

Reply:
A comment would also be nice, informing readers that `HttpHeaderRewriteFilter` has already verified it's either `http-request` or `http-response` 
(so readers don't have to verify that themselves, if they're worried it could be some other value)

Reply:
Nit: the parentheses are unnecessary (though if you find them more readable, that's fine)